device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -267,64 +291,91 @@ public void onConnectionEstablished(String connectionId)
+                this.updateStatus(IotHubConnectionStatus.DISCONNECTED_RETRYING, exceptionToStatusChangeReason(e), e, deviceId);
+
+                if (e instanceof TransportException)
+                {
+                    this.reconnectDeviceSession((TransportException) e, deviceId);
+                }
+                else
+                {
+                    this.reconnectDeviceSession(new TransportException(e), deviceId);
+                }
+            }
+        }
+    }
+
+    public void setMultiplexingRetryPolicy(RetryPolicy retryPolicy)
+    {
+        this.multiplexingRetryPolicy = retryPolicy;
+    }
+
     /**
      * Establishes a communication channel with an IoT Hub. If a channel is
      * already open, the function shall do nothing.
      *
      * If reconnection is occurring when this is called, this function shall block and wait for the reconnection
      * to finish before trying to open the connection
      *
-     * @param deviceClientConfigs the configs for the devices to open
-     *
      * @throws DeviceClientException if a communication channel cannot be
      * established.
      */
-    public void open(Collection<DeviceClientConfig> deviceClientConfigs) throws DeviceClientException
+    public void open() throws DeviceClientException
     {
-        if ((deviceClientConfigs == null) || deviceClientConfigs.isEmpty())
-        {
-            //Codes_SRS_IOTHUBTRANSPORT_34_015: [If the provided list of configs is null or empty, this function shall
-            // throw an IllegalArgumentException.]
-            throw new IllegalArgumentException("deviceClientConfigs cannot be null or empty");
-        }
-
         if (this.connectionStatus == IotHubConnectionStatus.CONNECTED)
         {
-            //Codes_SRS_IOTHUBTRANSPORT_34_017: [If the connection status of this object is CONNECTED, this function
-            // shall do nothing.]
             return;
         }
 
         if (this.connectionStatus == IotHubConnectionStatus.DISCONNECTED_RETRYING)
         {
-            //Codes_SRS_IOTHUBTRANSPORT_34_016: [If the connection status of this object is DISCONNECTED_RETRYING, this
-            // function shall throw a TransportException.]
             throw new TransportException("Open cannot be called while transport is reconnecting");
         }
 
-        if (this.isSasTokenExpired())
+        // The default config is only null when someone creates a multiplexing client and opens it before
+        // registering any devices to it. No need to check for SAS token expiry if no devices are registered yet.
+        if (this.getDefaultConfig() != null)
         {
-            //Codes_SRS_IOTHUBTRANSPORT_34_018: [If the saved SAS token has expired, this function shall throw a
-            // SecurityException.]
-            throw new SecurityException("Your sas token has expired");
+            if (this.isSasTokenExpired())
+            {
+                throw new SecurityException("Your sas token has expired");
```

Comment:
q - is sas token renewal handled in some other layer?

Reply:
To be honest, I don't know why this is here. It isn't very helpful, but I'll leave it alone just in case customers are catching this particular exception because we don't throw it anywhere else